libs/akita/src/__tests__/arrayToggle.spec.ts

```diff
@@ -0,0 +1,72 @@
+import { arrayToggle } from '../lib/arrayToggle';
+import { EntityStore } from '../lib/entityStore';
+import { StoreConfig } from '../lib/storeConfig';
+import { EntityState, ID } from '../lib/types';
+
+interface Comment {
+  id: ID;
+  text: string;
+}
+
+interface Article {
+  id: ID;
+  comments: Comment[];
+  title: string;
+}
+
+interface ArticlesState extends EntityState<Article> {
+  names: string[];
+}
+
+@StoreConfig({ name: 'articles' })
+class ArticlesStore extends EntityStore<ArticlesState, Article> {
+  constructor() {
+    super({ names: [] });
+  }
+}
+
+describe('arrayToggle', () => {
+  let store: ArticlesStore;
+
+  beforeEach(() => {
+    store = new ArticlesStore();
+  });
+
+  afterEach(() => {
+    store.destroy();
+  });
+
+  it('should add new item if none exist', () => {
+    store.add({
+      id: 1,
+      title: '',
+      comments: [],
+    });
+
+    store.update(1, (state) => {
+      return {
+        comments: arrayToggle(state.comments, { id: 1, text: 'comment' }, (a, b) => a.id === b.id),
+      };
+    });
+
+    expect(store._value().entities[1].comments.length).toBe(1);
+    expect(store._value().entities[1].comments[0].id).toBe(1);
+    expect(store._value().entities[1].comments[0].text).toBe('comment');
+  });
+
+  it('should remove item if one exist', () => {
+    store.add({
+      id: 1,
+      title: '',
+      comments: [{ id: 1, text: 'comment' }],
+    });
+
+    store.update(1, (state) => {
+      return {
+        comments: arrayToggle(state.comments, { id: 1, text: 'comment' }, (a, b) => a.id === b.id),
```

Comment:
byId is a common case, maybe we can expose helper function? `byId(idKey?)`:

`arrayToggle(state.comments, { id: 1, text: 'comment' }, byId()),`

Reply:
I added a byId() and byKey() comperator factory as you suggested.

Reply:
If it is omitted array entries are compared on identity as fallback